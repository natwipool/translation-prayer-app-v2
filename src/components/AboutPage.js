import React from 'react';

const AboutPage = () => (
  <div>
    <div className="page-header">
      <div className="content-container-body">
        <h2 className="page-header__title">เกี่ยวกับ</h2>
      </div>
    </div>
    <div className="content-container-body">
      <img className="cover-image" src="/images/about.jpg" />
      <div className="text-container">
        <h4>ขอขอบคุณ</h4>
        <p>
          ข้อมูลและไฟล์เสียงบทสวดมนต์จาก{' '}
          <a href="http://wimutti.org/" title="wimutti" target="_blank">
            เว็บไซต์ wimutti.org
          </a>
        </p>
        <p>
          Icons made by{' '}
          <a
              href="https://www.flaticon.com/authors/smashicons"
              title="Smashicons"
              target="_blank"
            >
              Smashicons,
            </a>{' '}
          <a
            href="https://www.flaticon.com/authors/zlatko-najdenovski"
            title="Zlatko Najdenovski"
            target="_blank"
          >
            Zlatko Najdenovski
          </a>{' '}
          from{' '}
          <a href="https://www.flaticon.com/" title="Flaticon" target="_blank">
            www.flaticon.com
          </a>{' '}
          is licensed by{' '}
          <a
            href="http://creativecommons.org/licenses/by/3.0/"
            title="Creative Commons BY 3.0"
            target="_blank"
          >
            CC 3.0 BY
          </a>
        </p>
        <br />
        <h4>ติดต่อสอบถามได้ที่</h4>
        <p>
          Facebook Page{' '}
          <a href="https://web.facebook.com/สวดมนต์-2068899220053295/">
            สวดมนต์
          </a>
        </p>
        <p>
          Email <a href="mailto:natwipool@gmail.com">natwipool@gmail.com</a>
        </p>
      </div>
    </div>
  </div>
);

export default AboutPage;
