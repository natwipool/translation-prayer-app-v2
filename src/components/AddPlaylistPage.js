import React from 'react';
import { connect } from 'react-redux';
import PlaylistForm from './PlaylistForm';
import { addPlaylist } from '../actions/playlists';

export class AddPlaylistPage extends React.Component {
  onSubmit = playlist => {
    this.props.addPlaylist(playlist);
    this.props.history.push('/playlists');
  };
  render() {
    return (
      <div>
        <h3>สร้างรายการสวดมนต์</h3>
        <PlaylistForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addPlaylist: playlist => dispatch(addPlaylist(playlist))
});

export default connect(undefined, mapDispatchToProps)(AddPlaylistPage);
