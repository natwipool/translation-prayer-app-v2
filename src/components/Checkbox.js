import React from 'react';

const Checkbox = props => {
  const onCheckboxChange = e => {
    const list = e.target.value;

    if (e.target.checked) {
      props.onAddLists(list);
    } else if (!e.target.checked) {
      props.onRemoveLists(list);
    }
  };

  return (
    <div>
      <label htmlFor={props.precept} className="checkbox">
        <input
          type="checkbox"
          id={props.precept}
          checked={props.lists.find(list => list === props.precept)}
          value={props.precept}
          onChange={onCheckboxChange}
        />
        <div className="checkbox-content">
        <span />
          
            <p className="checkbox-content__title">{props.precept}</p>
            
            {
              props.lists.indexOf(props.precept) > -1 &&
              <p className="checkbox-content__number">{props.lists.indexOf(props.precept) + 1}</p>
            } 
          
        </div>
      </label>
    </div>
  );
};

export default Checkbox;
