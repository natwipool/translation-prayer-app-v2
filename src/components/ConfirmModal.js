import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import { startRemovePlaylist } from '../actions/playlists';

export class ConfirmModal extends React.Component {
  onRemove = () => {
    this.props.startRemovePlaylist({ id: this.props.playlist.id });
    this.props.history.push('/playlists');
  };
  render() {
    return (
      <Modal
        isOpen={!!this.props.showConfirmModal}
        contentLabel="confirm-delete"
        onRequestClose={this.props.handleCloseModal}
        ariaHideApp={false}
        closeTimeoutMS={200}
        className="Modal"
      >
        <div className="confirm-modal-content">
          <h3 className="modal-confirm__title">ลบรายการ ?</h3>
          <h3 className="modal-confirm__title">
            {this.props.playlist.description}
          </h3>
        <div className="confirm-modal-group-button">
          <button
            className="button--secondary del-button"
            onClick={this.onRemove}
          >
            ตกลง
          </button>
          <button
            className="button--secondary edit-button"
            onClick={this.props.handleCloseModal}
          >
            ยกเลิก
          </button>
          </div>
        </div>
      </Modal>
    );
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  startRemovePlaylist: id => dispatch(startRemovePlaylist(id))
});

export default connect(undefined, mapDispatchToProps)(ConfirmModal);
