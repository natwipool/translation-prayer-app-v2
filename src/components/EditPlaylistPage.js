import React from 'react';
import { connect } from 'react-redux';
import PlaylistForm from './PlaylistForm';
import { startEditPlaylist } from '../actions/playlists';

export class EditPlaylistPage extends React.Component {
  onSubmit = playlist => {
    this.props.startEditPlaylist(this.props.playlist.id, playlist);
    this.props.history.push(`/playlist/${this.props.playlist.id}`);
  };
  render() {
    return (
      <div>
        <h3>แก้ไขรายการสวดมนต์</h3>
        <PlaylistForm playlist={this.props.playlist} onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  playlist: state.playlists.find(
    playlist => playlist.id === props.match.params.id
  )
});

const mapDispatchToProps = dispatch => ({
  startEditPlaylist: (id, playlist) => dispatch(startEditPlaylist(id, playlist))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditPlaylistPage);
