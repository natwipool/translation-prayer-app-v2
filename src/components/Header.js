import React from 'react';
import { connect } from 'react-redux';
import { Link, NavLink } from 'react-router-dom';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import LoginPage from '../components/LoginPage';
import { startLogout } from '../actions/auth';
import { handleShowLoginModal } from '../actions/modal';

export const Header = props => (
  <header>
    <Navbar inverse collapseOnSelect>
      <div className="content-container">
        <Navbar.Header>
          <Navbar.Brand>
            <Link to="/">สวดมนต์แปล</Link>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav>
            <LinkContainer to="/trans-prayers">
              <NavItem>สารบัญ</NavItem>
            </LinkContainer>
            <LinkContainer to="/playlists">
              <NavItem>บทสวดมนต์ของฉัน</NavItem>
            </LinkContainer>
            <LinkContainer to="/about">
              <NavItem>เกี่ยวกับ</NavItem>
            </LinkContainer>
          </Nav>
          <Nav pullRight>
            <NavItem>
              {!props.isAuthenticated ? (
                <button 
                  className="button header-button"
                  onClick={() => {
                    props.handleShowLoginModal(true);
                  }}
                >
                  เข้าสู่ระบบ
                </button>
              ) : (
                <button 
                  className="button header-button logout-button"
                  onClick={props.startLogout}
                >
                  ออกจากระบบ
                </button>
              )}
            </NavItem>
          </Nav>
        </Navbar.Collapse>
      </div>
    </Navbar>
    <LoginPage
      showLoginModal={!!props.loginModal}
      handleHideLoginModal={() => {
        props.handleShowLoginModal();
      }}
    />
  </header>
);

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid,
  loginModal: state.modal.showLoginModal
});

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
  handleShowLoginModal: isShow => dispatch(handleShowLoginModal(isShow))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
