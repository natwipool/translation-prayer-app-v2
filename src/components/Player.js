import React from 'react';
import { connect } from 'react-redux';
import ReactPlayer from 'react-player';
import {
  incrementIndex,
  decrementIndex,
  isPlayingToggle,
  setPlaying,
  setIndex
} from '../actions/players';
import formatTime from '../utils/format-time';

export class Player extends React.Component {
  state = {
    duration: undefined,
    currentTime: undefined,
    isReady: false
  };

  playlists = this.props.playlists.map(({ filename, precept }) => ({
    mp3: `https://s3-ap-southeast-1.amazonaws.com/transprayer/mp3/${filename}.mp3`,
    precept
  }));

  componentWillUnmount() {
    this.setState(() => ({ isReady: false }));
    this.props.setIndex();
    this.props.setPlaying();
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.players.index !== nextProps.players.index) {
      this.setState(() => ({ isReady: false }));
    }
  }

  isPlayingToggle = () => {
    console.log(this.props.players.index)
    this.props.isPlayingToggle();
    if (!this.props.players.isPlaying) {
      this.props.handleOpenModal();
    }
  };

  onNextClick = () => {
    this.props.incrementIndex();
  };

  onPrevClick = () => {
    this.props.decrementIndex();
  };

  onEndedEvent = () => {
    if (this.props.players.index + 1 === this.props.playlists.length) {
      this.props.setIndex();
      this.props.setPlaying();
    } else {
      this.props.incrementIndex();
      this.props.setPlaying(true);
    }
  };

  onReady = () => {
    this.setState(() => ({ isReady: true }));
  };

  onProgress = state => {
    this.setState(() => ({ currentTime: state.playedSeconds }));
  };

  onDuration = duration => {
    this.setState(() => ({ duration }));
  };

  ref = player => {
    this.player = player;
  };

  render() {
    return (
      <div className="player-container">
        <div className="player">
          <div>
            <ReactPlayer
              ref={this.ref}
              playing={this.props.players.isPlaying}
              url={this.playlists[this.props.players.index].mp3}
              onEnded={this.onEndedEvent}
              onProgress={this.onProgress}
              onDuration={this.onDuration}
              onReady={this.onReady}
              width="100%"
              height="100%"
            />
            <button
              className="music-button"
              onClick={this.onPrevClick}
              disabled={this.props.players.index === 0}
            >
              <img src="/images/prev.png" />
            </button>
            <button className="music-button" onClick={this.isPlayingToggle}>
              {this.props.players.isPlaying ? (
                <img src="/images/pause.png" />
              ) : (
                <img src="/images/play.png" />
              )}
            </button>
            <button
              className="music-button"
              onClick={this.onNextClick}
              disabled={this.props.players.index === this.playlists.length - 1}
            >
              <img src="/images/next.png" />
            </button>
          </div>
          <div className="player-content">
            {this.playlists && this.state.isReady ? (
              <h3 className="player-title move-text">
                {this.playlists[this.props.players.index].precept}
              </h3>
            ) : (
              <h3 className="player-title">Loading...</h3>
            )}
            <p className="player-timer">
              {this.state.currentTime !== undefined && this.state.isReady
                ? formatTime(this.state.currentTime)
                : '0.00'}
              {' / '}
              {this.state.duration && this.state.isReady
                ? formatTime(this.state.duration)
                : '0.00'}
            </p>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  players: state.players
});

const mapDispatchToProps = (dispatch, props) => ({
  incrementIndex: () => dispatch(incrementIndex()),
  decrementIndex: () => dispatch(decrementIndex()),
  isPlayingToggle: () => dispatch(isPlayingToggle()),
  setPlaying: () => dispatch(setPlaying()),
  setIndex: index => dispatch(setIndex(index))
});

export default connect(mapStateToProps, mapDispatchToProps)(Player);
