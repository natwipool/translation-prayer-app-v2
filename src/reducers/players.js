const defaultStatePlayerReducer = {
  isPlaying: false,
  index: 0
};

export default (state = defaultStatePlayerReducer, action) => {
  switch (action.type) {
    case 'IS_PLAYING_TOGGLE':
      return {
        ...state,
        isPlaying: !state.isPlaying,
      };
    case 'SET_PLAYING':
      return {
        ...state,
        isPlaying: true
      }
    case 'INCREMENT_INDEX':
      return {
        ...state,
        index: state.index + 1
      };
    case 'DECREMENT_INDEX':
      return {
        ...state,
        index: state.index - 1
      };
    case 'SET_INDEX':
      return {
        ...state,
        index: action.index
      }
    default:
      return state;
  }
};