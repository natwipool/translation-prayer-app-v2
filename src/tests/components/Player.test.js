import React from 'react';
import { shallow } from 'enzyme';
import { Player } from '../../components/Player';
import transPrayers from '../fixtures/transPrayers';
import players from '../fixtures/players';

let isPlayingToggle,
incrementIndex,
decrementIndex,
wrapper;

beforeEach(() => {
  isPlayingToggle = jest.fn();
  incrementIndex = jest.fn();
  decrementIndex = jest.fn();

  wrapper = shallow(<Player
    isPlayingToggle={isPlayingToggle}
    incrementIndex={incrementIndex}
    decrementIndex={decrementIndex}
    players={players.notPlaying}
    playlists={transPrayers}
  />);
});

test('should render Player correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onTogglePlaying correctly', () => {
  wrapper.find('button').at(1).simulate('click');
  expect(isPlayingToggle).toHaveBeenCalled();
});

test('should handle onNextClick correctly', () => {
  wrapper.find('button').at(2).simulate('click');
  expect(incrementIndex).toHaveBeenCalled();
});

test('should handle onPrevClick correctly', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(decrementIndex).toHaveBeenCalled();
});



