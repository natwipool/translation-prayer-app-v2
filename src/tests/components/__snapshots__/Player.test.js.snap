// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Player correctly 1`] = `
<div
  className="player-container"
>
  <ProgressBar
    active={false}
    bsClass="progress-bar"
    bsStyle="warning"
    isChild={false}
    max={100}
    min={0}
    srOnly={false}
    striped={false}
  />
  <div
    className="player"
  >
    <div>
      <ReactPlayer
        config={
          Object {
            "dailymotion": Object {
              "params": Object {
                "api": 1,
                "endscreen-enable": false,
              },
              "preload": false,
            },
            "facebook": Object {
              "appId": "1309697205772819",
            },
            "file": Object {
              "attributes": Object {},
              "forceAudio": false,
              "forceDASH": false,
              "forceHLS": false,
              "hlsOptions": Object {},
              "tracks": Array [],
            },
            "mixcloud": Object {
              "options": Object {
                "hide_cover": 1,
              },
            },
            "soundcloud": Object {
              "options": Object {
                "buying": false,
                "download": false,
                "liking": false,
                "sharing": false,
                "show_comments": false,
                "show_playcount": false,
                "visual": true,
              },
            },
            "vimeo": Object {
              "playerOptions": Object {
                "autopause": false,
                "autoplay": false,
                "byline": false,
                "portrait": false,
                "title": false,
              },
              "preload": false,
            },
            "wistia": Object {
              "options": Object {},
            },
            "youtube": Object {
              "playerVars": Object {
                "autoplay": 0,
                "iv_load_policy": 3,
                "modestbranding": 1,
                "playsinline": 1,
                "rel": 0,
                "showinfo": 0,
              },
              "preload": false,
            },
          }
        }
        controls={false}
        height="100%"
        loop={false}
        muted={false}
        onBuffer={[Function]}
        onDuration={[Function]}
        onEnded={[Function]}
        onError={[Function]}
        onPause={[Function]}
        onPlay={[Function]}
        onProgress={[Function]}
        onReady={[Function]}
        onSeek={[Function]}
        onStart={[Function]}
        playbackRate={1}
        playing={false}
        playsinline={false}
        progressInterval={1000}
        style={Object {}}
        url="https://s3-ap-southeast-1.amazonaws.com/transprayer/mp3/11_เขมาเขมสรณคมนปริทีปิกาคาถา.mp3.mp3"
        volume={0.8}
        width="100%"
        wrapper="div"
      />
      <button
        className="music-button"
        disabled={true}
        onClick={[Function]}
      >
        <img
          src="/images/prev.png"
        />
      </button>
      <button
        className="music-button"
        onClick={[Function]}
      >
        <img
          src="/images/play.png"
        />
      </button>
      <button
        className="music-button"
        disabled={false}
        onClick={[Function]}
      >
        <img
          src="/images/next.png"
        />
      </button>
    </div>
    <div
      className="player-content"
    >
      <h3
        className="player-title"
      >
        Loading...
      </h3>
      <p
        className="player-timer"
      >
        0.00
         / 
        0.00
      </p>
    </div>
  </div>
</div>
`;
